File: SkateboardAS/Core/Interfaces/IAssemblyService.cs
using Core.Models;

namespace Core.Interfaces;

public interface IAssemblyService
{
    Task<CommandResult> StartOperationAsync(string processId, CancellationToken ct = default);
    Task<bool> CheckHealthAsync(CancellationToken ct = default);
    Task<AssemblyStatus> GetAssemblyStatusAsync(CancellationToken ct = default);
}

File: SkateboardAS/core/src/main/java/dk/sdu/core/api/IAssemblyService.java
package dk.sdu.core.api;

import dk.sdu.core.model.AssemblyStatus;
import dk.sdu.core.model.CommandResult;

public interface IAssemblyService {
    CommandResult startOperation(String processId);
    boolean checkHealth();
    AssemblyStatus getAssemblyStatus();
}
